test/test-Heartbeat.js

```diff
@@ -0,0 +1,167 @@
+const tabs = require("sdk/tabs");
+const testRunner = require('sdk/test');
+const {before, after} = require('sdk/test/utils');
+
+const {Heartbeat} = require('../lib/Heartbeat.js');
+
+
+let targetWindow;
+let notificationBox;
+
+exports['test it shows a heartbeat panel'] = assert => {
+  assert.equal(notificationBox.childElementCount, 0);
+  new Heartbeat(targetWindow, {testing: true});
+  assert.equal(notificationBox.childElementCount, 1);
+};
+
+exports['test it shows five stars when there is no engagementButtonLabel'] = assert => {
+  let hb = new Heartbeat(targetWindow, {
+    testing: true,
+    engagementButtonLabel: undefined,
+  });
+  assert.equal(hb.notice.querySelectorAll('.star-x').length, 5);
+  assert.equal(hb.notice.querySelectorAll('.notification-button').length, 0);
+};
+
+exports['test it shows a button when there is an engagementButtonLabel'] = assert => {
+  let hb = new Heartbeat(targetWindow, {
+    testing: true,
+    engagementButtonLabel: 'Click me!',
+  });
+  assert.equal(hb.notice.querySelectorAll('.star-x').length, 0);
+  assert.equal(hb.notice.querySelectorAll('.notification-button').length, 1);
+};
+
+exports['test it shows a learn more link'] = assert => {
+  let hb = new Heartbeat(targetWindow, {
+    testing: true,
+    learnMoreMessage: 'Learn More',
+    learnMoreURL: 'https://example.org/learnmore',
+  });
+  let learnMoreEl = hb.notice.querySelector('.text-link');
+  assert.equal(learnMoreEl.href, 'https://example.org/learnmore');
+  assert.equal(learnMoreEl.value, 'Learn More');
+};
+
+exports['test it shows the message'] = assert => {
+  let hb = new Heartbeat(targetWindow, {
+    testing: true,
+    message: 'oh hai',
+  });
+  let messageEl = targetWindow.document.getAnonymousElementByAttribute(
+    hb.notice,
+    'anonid',
+    'messageText'
+  );
+  assert.equal(messageEl.textContent, 'oh hai');
+};
+
+exports['test it pings telemetry'] = (assert, done) => {
+  let hb = new Heartbeat(targetWindow, {
+    testing: true,
+    message: 'oh hai',
+  });
+
+  hb.events.on('TelemetrySent', payload => {
+    assert.ok(isOrdered([0, payload.offeredTS, payload.closedTS]));
+    done();
+  });
+
+  // triggers sending ping to normandy
+  closeAllNotifications();
+};
+
+exports['test it includes learnMoreTS if learn more is clicked'] = (assert, done) => {
+  let hb = new Heartbeat(targetWindow, {
+    testing: true,
+    message: 'oh hai',
+    learnMoreMessage: 'Learn More',
+    learnMoreURL: 'https://example.org/learnmore',
+  });
+
+  hb.events.on('TelemetrySent', payload => {
+    assert.ok(isOrdered([0, payload.offeredTS, payload.learnMoreTS, payload.closedTS]));
+    // Close learn more tab
+    tabs[tabs.length - 1].close();
+    done();
+  });
+
+  let learnMoreEl = hb.notice.querySelector('.text-link');
+  learnMoreEl.click();
+
+  // triggers sending ping to normandy
+  closeAllNotifications();
+};
+
+//exports['test it opens an engagement page after interaction'] = assert => {
```

Comment:
I had to change `tabs.on('open', ...)` to `tabs.on('load', ...)`. On `open`, the tab had the url of `about:blank`. Changing that and adding `www.` to the assertion line made this work great.